### PR TITLE
Update test_MarkovModel.py

### PR DESCRIPTION
**Pull request by MaxGreil**
Updated Tests/test_MarkovModel.py to expand unittest coverage.
See biopython issue #820

### CODE REVIEW

File: Tests/test_MarkovModel.py [hunk 1]
```diff
@@ -20,6 +23,11 @@
     warnings.simplefilter("ignore", UserWarning)
     from Bio import MarkovModel
 
+try:
```
Comment (peterjc) [inline Tests/test_MarkovModel.py]: Can you use just ``from Bio._py3k import StringIO`` here?

It is shorter, and will make it slightly easier to remove all the Python 2 specific code one day.

File: Tests/test_MarkovModel.py [hunk 2]
```diff
@@ -234,6 +242,254 @@ def test_topcoder5(self):
         state_list, state_float = states[0]
         self.assertEqual(state_list, ["N"])
 
+# Copyright 2017 by Maximilian Greil. All rights reserved.
```
Comment (peterjc) [inline Tests/test_MarkovModel.py]: Just put a single (combined) copyright notice at the top of the file please.

File: Tests/test_MarkovModel.py [hunk 3]
```diff
@@ -234,6 +242,254 @@ def test_topcoder5(self):
...
+        self.assertTrue(array_equal(
+            around(markov_model.p_initial, decimals=3),
+            around(output_p_initial, decimals=3)))
+        self.assertTrue(array_equal(around(
+            markov_model.p_transition, decimals=3),
+            around(output_p_transition, decimals=3)))
+        self.assertTrue(
+            array_equal(around(
+                        markov_model.p_emission, decimals=3),
+                        around(output_p_emission, decimals=3)))
```
Comment (peterjc) [inline Tests/test_MarkovModel.py]: The indentation of the first example is the most readable, with the two things being compared on their on lines (self contained).

File: Tests/test_MarkovModel.py [hunk 4]
```diff
@@ -234,6 +242,254 @@ def test_topcoder5(self):
...
+        states = ["CP", "IP"]
+        outputs = [2, 1, 0]
+        lp_initial = log([1.0, 0.0000001])
+        lp_transition = log([[0.7, 0.3], [0.5, 0.5]])
+        lp_emission = log([[0.6, 0.1, 0.3], [0.1, 0.7, 0.2]])
+
+        matrix = array([[0., -1.5606477, -3.07477539, -3.84932984],
+                        [-16.11809565, -2.4079455, -3.27544608, -4.5847794]])
+        self.assertTrue(
+            array_equal(around(MarkovModel._forward(len(states),
```
Comment (peterjc) [inline Tests/test_MarkovModel.py]: As above, put the ``array_equal`` on the first line so that the two things being compared is clearer.

**Comment by peterjc** [comment]
I don't know why the coverage report hasn't appeared on the pull request (yet), but this looks good. I've only spotted some minor style issues, and placement of copyright header.

**Comment by MaxGreil** [comment]
Hello Peter,
I tried to update my code according to your suggestions.
Do you think that it is now acceptable?

**Comment by MaxGreil** [comment]
Hello Peter,
I corrected all your mentioned style issues and the placement of the copyright header.

**Comment by peterjc** [comment]
I filled in the missing copyright information with https://github.com/biopython/biopython/commit/fd4f94a6e364b80ff16b1370c7c7b4408152cd51 and dual-licensed it with https://github.com/biopython/biopython/commit/6b0471310b0dbfde41cde76a96b75c67336af1ce (see #898).

Thanks for working on this Max :)

**Pull request merged**